icechunk/src/store.rs

```diff
@@ -561,10 +561,29 @@ async fn set_array_meta(
     array_meta: ArrayMetadata,
     repo: &mut Session,
 ) -> Result<(), StoreError> {
-    if repo.get_array(&path).await.is_ok() {
-        // TODO: we don't necessarily need to update both
-        repo.set_user_attributes(path.clone(), array_meta.attributes).await?;
-        repo.update_array(path, array_meta.zarr_metadata).await?;
+    if let Ok(node) = repo.get_array(&path).await {
+        // Check if the user attributes are different, if they are update them
+        let existing_attrs = match node.user_attributes {
+            None => None,
+            Some(UserAttributesSnapshot::Inline(atts)) => Some(atts),
+            // FIXME: implement
+            Some(UserAttributesSnapshot::Ref(_)) => None,
```

Comment:
Does this ever happen? For now we just fallback to original impl, we update

Reply:
It's not implemented yet, so no, it won't happen